main.go

```diff
@@ -17,6 +23,92 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+type MigrationRunner struct {
+	path string
+}
+
+func NewMigrationRunner(path string) *MigrationRunner {
+	return &MigrationRunner{path: path}
+}
+
+func (m *MigrationRunner) Run(ctx context.Context, conn *sql.Conn, lastRanId int) error {
+	files, err := m.loadMigrationFiles()
+	if err != nil {
+		return err
+	}
+
+	sort.Slice(files, func(i, j int) bool {
+		id1, err := extractMigrationID(files[i])
+		if err != nil {
+			log.Fatal(err)
+		}
+		id2, err := extractMigrationID(files[j])
+		if err != nil {
+			log.Fatal(err)
```

Comment:
Make sure to return errors so that the caller can decide what to do with them. `log.Fatal` will halt the execution of the program. This would make it really hard to do something like print a helpful error message back to the user when we encounter this error. It also makes it difficult to test this method.

Reply:
how would I return an error inside of the slice call since it is within a function that only has a boolean return value?